src/middlewared/middlewared/plugins/jail.py

```diff
@@ -482,3 +483,60 @@ def _import(self, job, jail):
         IOCImage().import_jail(jail)
 
         return True
+
+    @accepts()
+    def get_plugin_versions(self):
+        """
+        Fetches a list of pkg's from the http://pkg.cdn.trueos.org/iocage/
+        repo and returns a dictionary for each plugin
+        """
+        try:
+            pkgs = self.middleware.call_sync('cache.get', 'iocage_plugin_pkgs')
+
+            return pkgs
+        except Exception:
+            pass  # It's either new or past cache date
+
+        r_pkgs = requests.get('http://pkg.cdn.trueos.org/iocage/All')
```

Comment:
You should do `raise_for_status()` to check HTTP return codes, otherwise you may be parsing error 404 text